src/main/java/org/myrobotlab/service/abstracts/AbstractMotorController.java

```diff
@@ -1,17 +1,224 @@
 package org.myrobotlab.service.abstracts;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
 import org.myrobotlab.framework.Service;
+import org.myrobotlab.framework.interfaces.Attachable;
 import org.myrobotlab.math.Mapper;
+import org.myrobotlab.service.Motor;
+import org.myrobotlab.service.Runtime;
+import org.myrobotlab.service.interfaces.MotorControl;
 import org.myrobotlab.service.interfaces.MotorController;
 
 public abstract class AbstractMotorController extends Service implements MotorController {
+  
+  protected transient Map<String, AbstractMotor> motors = new HashMap<String, AbstractMotor>();
 
   private static final long serialVersionUID = 1L;
   
-  protected Mapper powerMapper = null;
+  // @Deprecated // use MotorControl's powerMapper as its "per" Motor ! - this is global :P
+  // protected Mapper powerMapper = null;
+  
+  /**
+   * default mapping and limits !
+   * This will be overriden by the AbstractMotor's values (if set)
+   */
+  // input range
+  Double minX = -1.0;
+  Double maxX = 1.0;
+
+  // output range
+  Double minY = -1.0;
+  Double maxY = 1.0;
+  
+  // min max of input
+  Double minInput = null;
+  Double maxInput = null;
+
+  // min max of output
+  Double minOutput = null;
+  Double maxOutput = null;
 
   public AbstractMotorController(String reservedKey) {
     super(reservedKey);    
   }
+  
+  // setting default map values
+  public void map(double minX, double maxX, double minY, double maxY) {
+    this.minX = minX;
+    this.maxX = maxX;
+    this.minY = minY;
+    this.maxY = maxY;
+    broadcastState();
+  }
+  
+  // TODO - switch to MotorControl interface instead of AbstractMotor
+  public double calcOutput(AbstractMotor mc) {
```

Comment:
All of this logic should probably move to the Mapper object..  also, we should probably pass the interface here, not the abstract...